go/arrow/scalar/scalar.go

```diff
@@ -664,6 +671,28 @@ func GetScalar(arr arrow.Array, idx int) (Scalar, error) {
 		return NewTime64Scalar(arr.Value(idx), arr.DataType()), nil
 	case *array.Timestamp:
 		return NewTimestampScalar(arr.Value(idx), arr.DataType()), nil
+	case *array.RunEndEncoded:
+		physicalIndex := encoded.FindPhysicalIndex(arr.Data(), arr.Offset()+idx)
+		value, err := GetScalar(arr.Values(), physicalIndex)
+		if err != nil {
+			return nil, err
+		}
+
+		var runLength int64
+
+		runEnds := arr.RunEndsArr()
+		switch ends := runEnds.(type) {
+		case *array.Int16:
+			runLength = int64(ends.Value(physicalIndex))
+		case *array.Int32:
+			runLength = int64(ends.Value(physicalIndex))
+		case *array.Int64:
+			runLength = int64(ends.Value(physicalIndex))
+		}
```

Comment:
Apologies if this discussion was already hashed out, but does it make sense to have a run-end-encoded scalar? Wouldn't 'scalar at logical index N' just be the scalar of the underlying type?

Reply:
I suppose at least here, we do adjust the runLength for the logical offset

Reply:
@felipecrv did you discuss the run-end-encoded scalar with @pitrou or anyone else? @lidavidm makes a good point that there's the possibility that we don't actually need a specific RunEndEncoded scalar type.

Reply:
To me, this scalar makes sense as representing the physical value for a physical index, but not the logical value for a logical index (if that makes...logical sense?)

Reply:
Of course, Arrow hasn't really differentiated like that in the past since effectively everything was a 'physical' value (even if there was sort of not really a distinction between physical and logical types). But dictionary and run-end encoding kind of changes that up (the latter more than the former), and what happens with StringView when we get there?

Reply:
@zeroshade Antoine has suggested that I add a RunEndEncoded scalar type instead of adding hacks to make things work without it.

> To me, this scalar makes sense as representing the physical value for a physical index, but not the logical value for a logical index (if that makes...logical sense?)

If you look at the REE array as an array of runs, the scalar that is typed by the REE type is a single run.